hepcrawl/spiders/edp_spider.py

```diff
@@ -0,0 +1,430 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2016 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Spider for EDP Sciences."""
+
+from __future__ import absolute_import, print_function
+
+import os
+import urlparse
+
+import tarfile
+
+from scrapy import Request
+from scrapy.spiders import XMLFeedSpider
+
+from ..extractors.jats import Jats
+from ..items import HEPRecord
+from ..loaders import HEPLoader
+from ..utils import (
+    ftp_list_files,
+    ftp_connection_info,
+    get_first,
+    get_node,
+    parse_domain,
+)
+
+
+class EDPSpider(Jats, XMLFeedSpider):
+    """EDP Sciences crawler.
+
+    This spider connects to a given FTP hosts and downloads zip files with
+    XML files for extraction into HEP records.
+
+    This means that it generates the URLs for Scrapy to crawl in a special way:
+
+    1. First it connects to a FTP host and lists all the new TAR files found
+       on the remote server and downloads them to a designated local folder,
+       using `start_requests()`. The starting point of the crawl can also be a
+       local file. Packages contain XML files with different formats (gz package
+       is JATS, bz2 package has "rich" and "jp" format XML files, "jp" is JATS.)
+
+    2. Then the TAR file is unpacked and it lists all the XML files found
+       inside, via `handle_package()`. Note the callback from `start_requests()`
+
+    3. Each XML file is parsed via `parse_node()`.
+
+    4. PDF file is fetched from the web page in `scrape_for_pdf()`.
+
+    5. Finally, HEPRecord is created in `build_item()`.
+
+    To run a crawl, you need to pass FTP connection information via
+    `ftp_host` and `ftp_netrc`:``
+
+    .. code-block:: console
+
+        scrapy crawl EDP -a ftp_host=ftp.edpsciences.org -a ftp_netrc=tmp/edps_netrc
+        scrapy crawl EDP -a package_path=file://`pwd`/tests/responses/edp/test_jp.tar.bz2
```

Comment:
The test_jp file does not exist, should it be the `file://$PWD/tests/responses/edp/test_package.tar.bz2`?

That one seems empty though, it would be nice if you cod use something with more content (looking to see if I can use an existing fixture or something).